lib/outh/google-outh.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const GoogleAuth = require('google-auth-library');
+const promisify = require('micro-promisify');
+const readlineSync = require('readline-sync');
+
+const { getMessage } = require('../utils/messages');
+
+import { AuthorizeCredentials } from '../../types/types';
+
+const fsReadFile = promisify(require('fs').readFile);
+const fsWriteFile = promisify(require('fs').writeFile);
+
+/* improve the bad polyfill that devtools-frontend did */
+//@todo remove after https://github.com/GoogleChrome/lighthouse/issues/1535 will be closed
+const globalAny:any = global;
+const self = globalAny.self || this;
+self.setImmediate = function(callback:any) {
+  Promise.resolve().then(_ => callback.apply(null, [...arguments].slice(1)));
+  return 0;
+};
+
+const EEXIST = 'EEXIST';
+
+class GoogleOuth {
+  scopes: Array<string>;
+  tokenDir: string;
+  tokenPath: string;
+
+  constructor() {
+    // If modifying these this.scopes, delete your previously saved credentials
+    // at ~/.credentials/sheets.googleapis.com-nodejs-pwmetrics.json
+    this.scopes = ['https://www.googleapis.com/auth/spreadsheets'];
```

Comment:
extract these strings to file level consts, also you can do these assignments on lines 30 - 32